### PR TITLE
Parser

### PR DESCRIPTION
**Pull request by WaterIris**

### CODE REVIEW

File: modules/framework/parser.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import pandas as pd 
+import numpy as np
```
Comment (skomposzczet) [inline modules/framework/parser.py:2]: add added python packages to requirements.txt

File: modules/framework/parser.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+from modules.db.model.car import Car
+
+
+class Return_Car:
```
Comment (skomposzczet) [inline modules/framework/parser.py]: change name to better reflect what class does (and use camel case) e.g. DataProvider or DataIterator

File: modules/framework/parser.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    def __init__(self):
+        df = pd.read_csv(filepath_or_buffer="./data/Cars in Movies.csv",
```
Comment (skomposzczet) [inline modules/framework/parser.py]: path to data file should be passed as an argument to \_\_init\_\_ function

File: modules/framework/parser.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                         delimiter=";",
+                         index_col= None)
+        self.df = df[["Car Full Name","Movie Title"]]
+        self.s_ind = 0
+        self.e_ind = 0
+
+    def gimmie_amount(self,amount):
```
Comment (skomposzczet) [inline modules/framework/parser.py]: rename function name to be more professional and better reflect what is does e.g.
fetch_n_rows or get_rows

File: modules/framework/parser.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        #start_time = timeit.default_timer()
```
Comment (skomposzczet) [inline modules/framework/parser.py]: remove unnecessary code

File: modules/framework/parser.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        self.e_ind+=amount
+        if self.s_ind == self.df.shape[0]:
+            return np.array([],dtype=Car)
+
+        if self.e_ind > self.df.shape[0]:
+            self.e_ind = self.df.shape[0]
+            amount = self.df.shape[0] - self.s_ind
+        else :
```
Comment (skomposzczet) [inline modules/framework/parser.py]: remove empty else clause

File: modules/framework/parser.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            pass
+
+        print(f"indexes {self.s_ind} : {self.e_ind-1}")
+        dt = np.empty(shape = amount,dtype=Car)
+        d = self.df.iloc[self.s_ind:self.e_ind,:]
+        n = 0
+        for index, row in d.iterrows():
```
Comment (skomposzczet) [inline modules/framework/parser.py]: if index variable is not used it should be name _ to indicate that its value is not used

File: modules/framework/parser.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+            sample = Car(full_name = row["Car Full Name"],movie_title=row["Movie Title"]) 
+            dt[n] = sample
+            n+=1
+        
+        self.s_ind = self.e_ind
+        #print(f"Time taken : {timeit.default_timer() - start_time:<.1f}s")
```
Comment (skomposzczet) [inline modules/framework/parser.py]: unnecessary code

File: modules/framework/parser.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+        df = pd.read_csv(filepath_or_buffer="./data/Cars in Movies.csv",
+                         delimiter=";",
+                         index_col= None)
+        self.df = df[["Car Full Name","Movie Title"]]
+        self.s_ind = 0
+        self.e_ind = 0
+
+    def gimmie_amount(self,amount):
+        #start_time = timeit.default_timer()
+        self.e_ind+=amount
```
Comment (skomposzczet) [inline modules/framework/parser.py]: does e_ind needs to be class field?
it seems that at beginning of every function call self.s_ind and self.e_ind are equal so e_ind could exist as local variable:
e_ind = self.s_ind + amount
and at the end of function
self.s_ind = e_ind
could work the same

File: modules/framework/parser.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+            amount = self.df.shape[0] - self.s_ind
+        else :
+            pass
+
+        print(f"indexes {self.s_ind} : {self.e_ind-1}")
+        dt = np.empty(shape = amount,dtype=Car)
+        d = self.df.iloc[self.s_ind:self.e_ind,:]
+        n = 0
+        for index, row in d.iterrows():
+            sample = Car(full_name = row["Car Full Name"],movie_title=row["Movie Title"])
```
Comment (skomposzczet) [inline modules/framework/parser.py]: try to obey to pep8 formatting standard e.g. remove trailing spaces, surround =+ operator with spaces, add space after comma etc.

**Review by skomposzczet — CHANGES_REQUESTED** [review]

File: modules/framework/parser.py (hunk 1 above)
Comment (WaterIris) [reply modules/framework/parser.py:2]: Done

**Review by skomposzczet — APPROVED** [review]

**Pull request merged**